client/normal_new.go

```diff
@@ -0,0 +1,465 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package client
+
+import (
+	"time"
+
+	"github.com/sourcenetwork/immutable"
+)
+
+// NewNormalValue creates a new NormalValue from the given value.
+// It will normalize all known types that can be converted to normal ones.
+// For example, is the given type is `[]int32`, it will be converted to `[]int64`.
+// If the given value is of type `[]any` is will to through every element and try to convert it
+// minimal common type and normalizes it.
+// For examples, the following conversions will be made:
+//   - `[]any{int32(1), int64(2)}` -> `[]int64{1, 2}`.
+//   - `[]any{int32(1), int64(2), float32(1.5)}` -> `[]float64{1.0, 2.0, 1.5}`.
+//   - `[]any{int32(1), nil}` -> `[]immutable.Option[int64]{immutable.Some(1), immutable.None[int64]()}`.
+//
+// This function will not check if the given value is `nil`. To normalize a `nil` value use the
+// `NewNormalNil` function.
+func NewNormalValue(val any) (NormalValue, error) {
```

Comment:
thought: If we passed `FieldKind` in here it would make it even more strongly typed given that `[]any{int32(1), int64(2)}` could very well be `[]immutable.Option[int64]{immutable.Some(1), immutable.Some(2)}` instead of `[]int64{1, 2}`. This can be done in a separate PR though.

Reply:
this is somewhat involved addition. I've created another task for it https://github.com/sourcenetwork/defradb/issues/2463